managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1603,91 +1603,94 @@ static ReadEntryCallbackWrapper create(String name, long ledgerId, long entryId,
             return readCallback;
         }
 
-        public boolean isTimedOut(long timeoutSec) {
-            return this.createdTime != -1
-                    && TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - this.createdTime) >= timeoutSec
-                    && this.readCompleted == FALSE;
-        }
-
         @Override
         public void readEntryComplete(Entry entry, Object ctx) {
-            if (checkCallbackCompleted(ctx)) {
+            long reOpCount = (ctx != null && ctx instanceof Long) ? (long) ctx : 0;
```

Comment:
I see this pattern is repeated many times. Can we abstract this logic into a common method or rewrite those methods using this pattern into one common method?

Reply:
sure.. made the change.